internal/clients/http/sources/sources_client.go

```diff
@@ -107,51 +115,54 @@ func (c *SourcesClient) ListProvisioningSources(ctx context.Context) (*[]clients
 }
 
 func (c *SourcesClient) GetArn(ctx context.Context, sourceId clients.ID) (string, error) {
-	ctxval.Logger(ctx).Info().Msgf("Getting ARN of source %v", sourceId)
+	c.logger.Trace().Msgf("Getting ARN of source %v", sourceId)
+
 	// Get all the authentications linked to a specific source
 	resp, err := c.client.ListSourceAuthenticationsWithResponse(ctx, sourceId, &ListSourceAuthenticationsParams{}, headers.AddSourcesIdentityHeader)
 	if err != nil {
 		return "", fmt.Errorf("cannot list source authentication: %w", err)
 	}
-	statusCode := resp.StatusCode()
-	if parsing.IsHTTPNotFound(statusCode) {
-		return "", NewNotFoundError(ctx, "authentications for source weren't found in sources app")
-	}
-	if !parsing.IsHTTPStatus2xx(statusCode) {
-		ctxval.Logger(ctx).Warn().Msgf("Sources replied with unexpected status while fetching Authentications per source: %v", statusCode)
-		return "", SourcesClientErr
+
+	err = http.HandleHTTPResponses(ctx, resp.StatusCode())
+	if err != nil {
+		if errors.Is(err, clients.NotFoundErr) {
+			return "", fmt.Errorf("get source ARN call: %w", SourceNotFoundErr)
+		}
+		return "", fmt.Errorf("get source ARN call: %w", err)
```

Comment:
What about just wrapping it?
```suggestion
		return "", fmt.Errorf("get source ARN call: %w", err)
```